json/src/main/java/com/basistech/rosette/dm/jackson/EntityMixin.java

```diff
@@ -0,0 +1,36 @@
+/*
+* Copyright 2014 Basis Technology Corp.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package com.basistech.rosette.dm.jackson;
+
+import com.basistech.rosette.dm.Mention;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@link com.basistech.rosette.dm.ResolvedEntity}
+ */
+public abstract class EntityMixin {
+    @JsonCreator
+    EntityMixin(@JsonProperty("entityId") String entityId,
+                @JsonProperty("mentions") List<Mention> mentions,
+                @JsonProperty("confidence") Double confidence,
+                @JsonProperty("extendedProperties") Map<String, Object> extendedProperties) {
+        //
```

Comment:
The chainId is currently defined to be the index of the head mention, so your
first two statements don't make sense to me.

The 'type' could be controversial.  The current ResolvedEntity does not have
a type, just an entityId.  The idea is that this id is sufficient for the user to 
lookup anything else about the entity in some external kb, including type.